// @nx/nest/src/generators/controller/controller.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.controllerGeneratorInternal = exports.controllerGenerator = void 0;
const utils_1 = require("../utils");
async function controllerGenerator(tree, rawOptions) {
    await controllerGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.controllerGenerator = controllerGenerator;
async function controllerGeneratorInternal(tree, rawOptions) {
    const options = await normalizeControllerOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'controller', options);
}
exports.controllerGeneratorInternal = controllerGeneratorInternal;
exports.default = controllerGenerator;
async function normalizeControllerOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'controller', '@nx/nest:controller', options);
    return {
        ...normalizedOptions,
        language: options.language,
        module: options.module,
        skipImport: options.skipImport,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
