// @nx/nest/src/generators/module/module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.moduleGeneratorInternal = exports.moduleGenerator = void 0;
const utils_1 = require("../utils");
async function moduleGenerator(tree, rawOptions) {
    await moduleGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.moduleGenerator = moduleGenerator;
async function moduleGeneratorInternal(tree, rawOptions) {
    const options = await normalizeModuleOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'module', options);
}
exports.moduleGeneratorInternal = moduleGeneratorInternal;
exports.default = moduleGenerator;
async function normalizeModuleOptions(tree, options) {
    const normalizedOption = await (0, utils_1.normalizeOptions)(tree, 'module', '@nx/nest:module', options);
    return {
        ...normalizedOption,
        language: options.language,
        module: options.module,
        skipImport: options.skipImport,
    };
}
