// @nx/nest/src/generators/gateway/gateway.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.gatewayGeneratorInternal = exports.gatewayGenerator = void 0;
const utils_1 = require("../utils");
async function gatewayGenerator(tree, rawOptions) {
    await gatewayGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.gatewayGenerator = gatewayGenerator;
async function gatewayGeneratorInternal(tree, rawOptions) {
    const options = await normalizeGatewayOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'gateway', options);
}
exports.gatewayGeneratorInternal = gatewayGeneratorInternal;
exports.default = gatewayGenerator;
async function normalizeGatewayOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'gateway', '@nx/nest:gateway', options);
    return {
        ...normalizedOptions,
        language: options.language,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
