// @nx/nest/src/generators/resource/resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.resourceGeneratorInternal = exports.resourceGenerator = void 0;
const utils_1 = require("../utils");
async function resourceGenerator(tree, rawOptions) {
    await resourceGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.resourceGenerator = resourceGenerator;
async function resourceGeneratorInternal(tree, rawOptions) {
    const options = await normalizeResourceOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'resource', options);
}
exports.resourceGeneratorInternal = resourceGeneratorInternal;
exports.default = resourceGenerator;
async function normalizeResourceOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'resource', '@nx/nest:resource', options);
    return {
        ...normalizedOptions,
        language: options.language,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
