// @nx/nest/src/generators/application/application.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.applicationGeneratorInternal = exports.applicationGenerator = void 0;
const devkit_1 = require("@nx/devkit");
const node_1 = require("@nx/node");
const init_1 = require("../init/init");
const lib_1 = require("./lib");
async function applicationGenerator(tree, rawOptions) {
    return await applicationGeneratorInternal(tree, {
        projectNameAndRootFormat: 'derived',
        ...rawOptions,
    });
}
exports.applicationGenerator = applicationGenerator;
async function applicationGeneratorInternal(tree, rawOptions) {
    const options = await (0, lib_1.normalizeOptions)(tree, rawOptions);
    const initTask = await (0, init_1.initGenerator)(tree, {
        skipPackageJson: options.skipPackageJson,
        unitTestRunner: options.unitTestRunner,
        skipFormat: true,
    });
    const nodeApplicationTask = await (0, node_1.applicationGenerator)(tree, (0, lib_1.toNodeApplicationGeneratorOptions)(options));
    (0, lib_1.createFiles)(tree, options);
    (0, lib_1.updateTsConfig)(tree, options);
    if (!options.skipFormat) {
        await (0, devkit_1.formatFiles)(tree);
    }
    return (0, devkit_1.runTasksInSerial)(initTask, nodeApplicationTask);
}
exports.applicationGeneratorInternal = applicationGeneratorInternal;
exports.default = applicationGenerator;
