// @nx/nest/src/generators/filter/filter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.filterGeneratorInternal = exports.filterGenerator = void 0;
const utils_1 = require("../utils");
async function filterGenerator(tree, rawOptions) {
    await filterGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.filterGenerator = filterGenerator;
async function filterGeneratorInternal(tree, rawOptions) {
    const options = await normalizeFilterOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'filter', options);
}
exports.filterGeneratorInternal = filterGeneratorInternal;
exports.default = filterGenerator;
async function normalizeFilterOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'filter', '@nx/nest:filter', options);
    return {
        ...normalizedOptions,
        language: options.language,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
