// @nx/webpack/src/executors/webpack/lib/normalize-options.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.normalizePluginPath = exports.normalizeOptions = void 0;
const path_1 = require("path");
const normalize_options_1 = require("../../../plugins/nx-webpack-plugin/lib/normalize-options");
function normalizeOptions(options, root, projectRoot, sourceRoot) {
    return {
        ...options,
        root,
        projectRoot,
        sourceRoot,
        target: options.target ?? 'web',
        outputFileName: options.outputFileName ?? 'main.js',
        assets: (0, normalize_options_1.normalizeAssets)(options.assets, root, sourceRoot),
        webpackConfig: normalizePluginPath(options.webpackConfig, root),
        optimization: typeof options.optimization !== 'object'
            ? {
                scripts: options.optimization,
                styles: options.optimization,
            }
            : options.optimization,
    };
}
exports.normalizeOptions = normalizeOptions;
function normalizePluginPath(pluginPath, root) {
    if (!pluginPath) {
        return '';
    }
    try {
        return require.resolve(pluginPath);
    }
    catch {
        return (0, path_1.resolve)(root, pluginPath);
    }
}
exports.normalizePluginPath = normalizePluginPath;
