// @nx/nest/src/generators/resolver/resolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.resolverGeneratorInternal = exports.resolverGenerator = void 0;
const utils_1 = require("../utils");
async function resolverGenerator(tree, rawOptions) {
    await resolverGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.resolverGenerator = resolverGenerator;
async function resolverGeneratorInternal(tree, rawOptions) {
    const options = await normalizeResolverOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'resolver', options);
}
exports.resolverGeneratorInternal = resolverGeneratorInternal;
exports.default = resolverGenerator;
async function normalizeResolverOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'resolver', '@nx/nest:resolver', options);
    return {
        ...normalizedOptions,
        language: options.language,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
