// @nx/nest/src/generators/service/service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.serviceGeneratorInternal = exports.serviceGenerator = void 0;
const utils_1 = require("../utils");
async function serviceGenerator(tree, rawOptions) {
    await serviceGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.serviceGenerator = serviceGenerator;
async function serviceGeneratorInternal(tree, rawOptions) {
    const options = await normalizeServiceOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'service', options);
}
exports.serviceGeneratorInternal = serviceGeneratorInternal;
exports.default = serviceGenerator;
async function normalizeServiceOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'service', '@nx/nest:service', options);
    return {
        ...normalizedOptions,
        language: options.language,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
