// @nx/nest/src/generators/utils/normalize-options.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unitTestRunnerToSpec = exports.normalizeOptions = void 0;
const artifact_name_and_directory_utils_1 = require("@nx/devkit/src/generators/artifact-name-and-directory-utils");
async function normalizeOptions(tree, artifactType, callingGenerator, options) {
    const { directory, fileName } = await (0, artifact_name_and_directory_utils_1.determineArtifactNameAndDirectoryOptions)(tree, {
        callingGenerator,
        artifactType,
        name: options.name,
        directory: options.directory,
        project: options.project,
        flat: options.flat,
        derivedDirectory: options.directory,
        nameAndDirectoryFormat: options.nameAndDirectoryFormat,
    });
    return {
        ...options,
        flat: true,
        name: fileName,
        skipFormat: options.skipFormat,
        sourceRoot: directory,
    };
}
exports.normalizeOptions = normalizeOptions;
function unitTestRunnerToSpec(unitTestRunner) {
    return unitTestRunner !== undefined ? unitTestRunner === 'jest' : undefined;
}
exports.unitTestRunnerToSpec = unitTestRunnerToSpec;
