// @nx/webpack/src/executors/dev-server/dev-server.impl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.devServerExecutor = void 0;
const webpack = require("webpack");
const devkit_1 = require("@nx/devkit");
const rxjs_for_await_1 = require("@nx/devkit/src/utils/rxjs-for-await");
const operators_1 = require("rxjs/operators");
const WebpackDevServer = require("webpack-dev-server");
const get_dev_server_config_1 = require("./lib/get-dev-server-config");
const buildable_libs_utils_1 = require("@nx/js/src/utils/buildable-libs-utils");
const run_webpack_1 = require("../../utils/run-webpack");
const custom_webpack_1 = require("../../utils/webpack/custom-webpack");
const normalize_options_1 = require("../webpack/lib/normalize-options");
const path_1 = require("path");
async function* devServerExecutor(serveOptions, context) {
    // Default to dev mode so builds are faster and HMR mode works better.
    process.env.NODE_ENV ??= 'development';
    const { root: projectRoot, sourceRoot } = context.projectsConfigurations.projects[context.projectName];
    const buildOptions = (0, normalize_options_1.normalizeOptions)(getBuildOptions(serveOptions, context), context.root, projectRoot, sourceRoot);
    if (!buildOptions.index) {
        throw new Error(`Cannot run dev-server without "index" option. Check the build options for ${context.projectName}.`);
    }
    if (!buildOptions.buildLibsFromSource) {
        const { target, dependencies } = (0, buildable_libs_utils_1.calculateProjectBuildableDependencies)(context.taskGraph, context.projectGraph, context.root, context.projectName, 'build', // should be generalized
        context.configurationName);
        buildOptions.tsConfig = (0, buildable_libs_utils_1.createTmpTsConfig)(buildOptions.tsConfig, context.root, target.data.root, dependencies);
    }
    let config = (0, get_dev_server_config_1.getDevServerConfig)(context, buildOptions, serveOptions);
    if (buildOptions.webpackConfig) {
        let tsconfigPath = buildOptions.tsConfig.startsWith(context.root)
            ? buildOptions.tsConfig
            : (0, path_1.join)(context.root, buildOptions.tsConfig);
        let customWebpack = (0, custom_webpack_1.resolveCustomWebpackConfig)(buildOptions.webpackConfig, tsconfigPath);
        if (typeof customWebpack.then === 'function') {
            customWebpack = await customWebpack;
        }
        if (typeof customWebpack === 'function') {
            // Old behavior, call the webpack function that is specific to Nx
            config = await customWebpack(config, {
                options: buildOptions,
                context,
                configuration: serveOptions.buildTarget.split(':')[2],
            });
        }
        else if (customWebpack) {
            // New behavior, use the config object as is with devServer defaults
            config = {
                devServer: {
                    ...customWebpack.devServer,
                    ...config.devServer,
                },
                ...customWebpack,
            };
        }
    }
    return yield* (0, rxjs_for_await_1.eachValueFrom)((0, run_webpack_1.runWebpackDevServer)(config, webpack, WebpackDevServer).pipe((0, operators_1.tap)(({ stats }) => {
        console.info(stats.toString(config.stats));
    }), (0, operators_1.map)(({ baseUrl, stats }) => {
        return {
            baseUrl,
            success: !stats.hasErrors(),
        };
    })));
}
exports.devServerExecutor = devServerExecutor;
function getBuildOptions(options, context) {
    const target = (0, devkit_1.parseTargetString)(options.buildTarget, context);
    const overrides = {
        watch: false,
    };
    if (options.memoryLimit) {
        overrides.memoryLimit = options.memoryLimit;
    }
    if (options.baseHref) {
        overrides.baseHref = options.baseHref;
    }
    const buildOptions = (0, devkit_1.readTargetOptions)(target, context);
    return {
        ...buildOptions,
        ...overrides,
    };
}
exports.default = devServerExecutor;
