// @nx/nest/src/generators/middleware/middleware.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.middlewareGeneratorInternal = exports.middlewareGenerator = void 0;
const utils_1 = require("../utils");
async function middlewareGenerator(tree, rawOptions) {
    await middlewareGeneratorInternal(tree, {
        nameAndDirectoryFormat: 'derived',
        ...rawOptions,
    });
}
exports.middlewareGenerator = middlewareGenerator;
async function middlewareGeneratorInternal(tree, rawOptions) {
    const options = await normalizeMiddlewareOptions(tree, rawOptions);
    return (0, utils_1.runNestSchematic)(tree, 'middleware', options);
}
exports.middlewareGeneratorInternal = middlewareGeneratorInternal;
exports.default = middlewareGenerator;
async function normalizeMiddlewareOptions(tree, options) {
    const normalizedOptions = await (0, utils_1.normalizeOptions)(tree, 'middleware', '@nx/nest:middleware', options);
    return {
        ...normalizedOptions,
        language: options.language,
        spec: (0, utils_1.unitTestRunnerToSpec)(options.unitTestRunner),
    };
}
